src/App.js

```diff
@@ -1,19 +1,43 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog.js';
+import { useState } from 'react';
+
+//const chats = chatMessages;
+
+function App() {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  // const calculateTotalLikes = (chatData) => {
+  //   const totalLikes = 0;
+  //   chatData.map((entry) => {
+  //     totalLikes += entry.liked;
+  //   });
+  //   return totalLikes;
+  // };
+
+  const updateChatData = (updatedChat) => {
+    const entries = chatData.map((entry) => {
+      if (entry.id === updatedChat.id) {
```

Comment:
The state is not properly being updated. In the ChatEntry component, only the key `liked` is passed in `updatedChat`. There is no `id`. Since there is no `id`, none of the entries will match the `updatedChat` and none of the messages will get updated in state here.